torch/csrc/distributed/c10d/ProcessGroup.cpp

```diff
@@ -5,6 +5,28 @@
 #include <fmt/format.h>
 
 namespace c10d {
+namespace {
+
+c10::intrusive_ptr<ProcessGroup::Work> broadcast(const c10::intrusive_ptr<ProcessGroup>& process_group,
+    at::TensorList tensors, int64_t root_rank, int64_t root_tensor, int64_t timeout) {
```

Comment:
Why is the signature of this broadcast method different from https://github.com/pytorch/pytorch/blob/master/torch/distributed/distributed_c10d.py#L1162?

Reply:
The parameters are unbundled from BroadcastOptions. `dist.broadcast` is a subset of this API.